### ISSUE TITLE
Error keeps popping up on Ventura

### ISSUE THREAD

**Issue by SmartDucksWorks** [cbreak-black/ZetaWatch#31]
Error Internal NV list. Check on several machines. Something about 'error - reading pool configuration'. Systems seem to be working fine, not seeing adverse effects but I am checking daily just in case.
![Screenshot 2023-06-03 at 9 18 31 AM](https://github.com/cbreak-black/ZetaWatch/assets/98773785/74b24651-2cc0-42d5-8226-1d3283a6a8be)

**Comment by cbreak-black** [comment]
This means that your ZetaWatch version does not match your ZFS version. This is very easily possible since I've not recompiled / updated it in ages. If possible, compile a ZetaWatch for your ZFS version (which version do you use?)

**Comment by SmartDucksWorks** [comment]
zfs-macOS-2.1.6-1
zfs-kmod-2.1.0-1
ZetaWatch-r50-0-g82304d-arm64.zip

**Comment by cbreak-black** [comment]
Hmm... your kmod and userland lib version don't match... that's possibly problematic. But that version of ZetaWatch was compiled for 2.1.6, so it is supposed to work I think.

**Comment by SmartDucksWorks** [comment]
Yeah, you are correct. Seems the installer from OpenZFSonOSX may be the culprit. Will uninstall manually and reinstall to see if that is indeed the actual issue. Should have caught that.

**Comment by SmartDucksWorks** [comment]
Tried compiling your code, that has more fun errors. ;) I

**Comment by cbreak-black** [comment]
Will have to reboot for you to be able to replace the kernel extension, in my experience, unloading it at runtime doesn't work reliably.

The code should compile... but I've only ever compiled it on my own mac, which is old and running an old MacOS and Xcode.

**Comment by SmartDucksWorks** [comment]
OK did a manual clean, reinstalled OpenZFSonOSX, and latest ZetaWatch binary for M! on 3 systems to test. 
zfs version
zfs-macOS-2.1.6-1
zfs-kmod-2.1.6-1

However the ZetaWatch is still providing the same error as before.

**Comment by cbreak-black** [comment]
Very weird :(

This error is detected here:
https://github.com/cbreak-black/ZFSWrapper/blob/b0ef0ebe812597676cf88d08a42bcc2be14e65b6/src/ZFSUtils.cpp#L1380-L1382

This code tries to read `ZPOOL_CONFIG_VDEV_STATS` and reinterpret it as `vdev_stat_t`, which I think is the expected way to do this. The test checks if the size of the structure is still correct.

**Comment by SmartDucksWorks** [comment]
So far I'm not seeing any actual issue, except for the error constantly popping up. Will keep evaluating as things progress.
